### PR TITLE
Add BmcAcpiSwChildPolicyProtocol and RedirFruProtocol definition

### PR DESCRIPTION
**Pull request by MarcChen46**
# Preface

Please ensure you have read the [contribution docs](https://github.com/microsoft/mu/blob/master/CONTRIBUTING.md) prior
to submitting the pull request. In particular,
[pull request guidelines](https://github.com/microsoft/mu/blob/master/CONTRIBUTING.md#pull-request-best-practices).

## Description

Add BmcAcpiSwChildPolicyProtocol and RedirFruProtocol definition and header files from edk2platform's IpmiFeaturePkg for compatible build with Intel Platform

For each item, place an "x" in between `[` and `]` if true. Example: `[x]`.
_(you can also check items in the GitHub UI)_

- [ ] Impacts functionality?
  - **Functionality** - Does the change ultimately impact how firmware functions?
  - Examples: Add a new library, publish a new PPI, update an algorithm, ...
- [ ] Impacts security?
  - **Security** - Does the change have a direct security impact on an application,
    flow, or firmware?
  - Examples: Crypto algorithm change, buffer overflow fix, parameter
    validation improvement, ...
- [ ] Breaking change?
  - **Breaking change** - Will anyone consuming this change experience a break
    in build or boot behavior?
  - Examples: Add a new library class, move a module to a different repo, call
    a function in a new library class in a pre-existing module, ...
- [ ] Includes tests?
  - **Tests** - Does the change include any explicit test code?
  - Examples: Unit tests, integration tests, robot tests, ...
- [ ] Includes documentation?
  - **Documentation** - Does the change contain explicit documentation additions
    outside direct code modifications (and comments)?
  - Examples: Update readme file, add feature readme file, link to documentation
    on an a separate Web page, ...

## How This Was Tested

Dummy code that does not impact any existed implementation.

## Integration Instructions

N/A

### CODE REVIEW

File: IpmiFeaturePkg/Include/Protocol/BmcAcpiSwChildPolicy.h [hunk 1]
```diff
@@ -0,0 +1,31 @@
+/** @file
+  This protocol produces BmcAcpiSwChildPolicy Protocol.
+
+Copyright (c) 2023, Intel Corporation. All rights reserved.<BR>
+SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+
+#ifndef _BMC_ACPI_SW_CHILD_POLICY_H_
```
Comment (apop5) [inline IpmiFeaturePkg/Include/Protocol/BmcAcpiSwChildPolicy.h:9]:
```suggestion
#ifndef BMC_ACPI_SW_CHILD_POLICY_H_

```

https://tianocore-docs.github.io/edk2-CCodingStandardsSpecification/release-2.20/5_source_files/53_include_files.html#53-include-files

Include guards that start with _ or __ are reserved for compilers. Can you please remove the _?

File: IpmiFeaturePkg/Include/Protocol/RedirFru.h [hunk 2]
```diff
@@ -0,0 +1,81 @@
+/** @file
+  This code abstracts the generic FRU Protocol.
+
+Copyright (c) 2023, Intel Corporation. All rights reserved.<BR>
+SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+
+#ifndef _REDIR_FRU_H_
```
Comment (apop5) [inline IpmiFeaturePkg/Include/Protocol/RedirFru.h:9]: Same with this include guard.

**Review by apop5 — APPROVED** [review]

File: IpmiFeaturePkg/Include/Protocol/BmcAcpiSwChildPolicy.h (hunk 1 above)
Comment (MarcChen46) [reply IpmiFeaturePkg/Include/Protocol/BmcAcpiSwChildPolicy.h]: updated

File: IpmiFeaturePkg/Include/Protocol/RedirFru.h (hunk 2 above)
Comment (MarcChen46) [reply IpmiFeaturePkg/Include/Protocol/RedirFru.h]: updated

**Review by os-d — APPROVED** [review]

**Pull request merged**